SocketRocket/SRWebSocket.m

```diff
@@ -1421,7 +1426,17 @@ - (void)stream:(NSStream *)aStream handleEvent:(NSStreamEvent)eventCode
         (eventCode == NSStreamEventHasBytesAvailable || eventCode == NSStreamEventHasSpaceAvailable)) {
         SecTrustRef trust = (__bridge SecTrustRef)[aStream propertyForKey:(__bridge id)kCFStreamPropertySSLPeerTrust];
         if (trust) {
-            _streamSecurityValidated = [_securityPolicy evaluateServerTrust:trust forDomain:_urlRequest.URL.host];
+            NSString *const host = _urlRequest.URL.host;
+            if (!host || host.length == 0) {
+                dispatch_async(_workQueue, ^{
+                    NSError *error = SRErrorWithDomainCodeDescription(NSURLErrorDomain,
+                                                                      NSURLErrorBadURL,
+                                                                      @"Unable to validate certificate for empty host.");
+                    [wself _failWithError:error];
+                });
+                return;
+            }
```

Comment:
nit: why do we need to jump thread to emit an error?

Reply:
:) Great question, we actually don't, you are right!
Let me clean it up after, there are more cases like this, the worst case - it's an extra dispatch here in error case.